include/DirectSolver.hpp

```diff
@@ -11,14 +11,15 @@
 #include "Types.hpp"
 
 namespace blitzdg {
-  class DirectSolver {
-      index_type N;
-      SparseMatrixConverter MatrixConverter;
+    class DirectSolver {
+        SparseMatrixConverter MatrixConverter;
   
-  public:
-      DirectSolver(SparseMatrixConverter const &);
+    public:
+        DirectSolver()
+        : MatrixConverter{}
```

Comment:
not quite familiar with this syntax in C++. Is this an initialization list? why do we need the {}'s? sorry for the dumb Questions...

Reply:
Yes, this is a C++ initialization list. It's best to initialize members this way if possible. The braces do brace-initialization, which offers some benefits over the usual bracket-initialization, i.e., (). Brace-initialization has the benefits of not allowing implicit copy construction (unlike =) or narrowing conversions. It does have a few caveats, but they are easy to avoid. I generally prefer it now in the code I write.

Reply:
By the way, your question isn't dumb at all buddy. I only started following this practice recently after learning about the benefits that it provides.